app/models/site_settings.rb

```diff
@@ -13,11 +13,29 @@
 #
 
 class SiteSettings < ApplicationRecord
+  store_accessor :prices, :electricity_price, :solar_export_price, :gas_price, :oil_price
+  validates :electricity_price, :solar_export_price, :gas_price, :oil_price, numericality: { only_float: true, allow_blank: true }
+  after_save :delete_current_prices_cache
+
+  CURRENT_PRICES_CACHE_KEY = 'site_settings_current_prices'.freeze
+
   def self.current
     order('created_at DESC').first || new
   end
 
   def temperature_recording_month_numbers
     temperature_recording_months.reject(&:blank?).map(&:to_i)
   end
+
+  def self.current_prices
+    Rails.cache.fetch(CURRENT_PRICES_CACHE_KEY) do
```

Comment:
What's the default cache time here? Should we be invalidating the cache when the SiteSettings are updated (if the prices have changed)

Reply:
no default cache time - we'll invalidate the cache after save (so runs both on create and update (just to be safe) - see `delete_current_prices_cache` l18)